src/Http/Action/Security/SsoRedirectAction.php

```diff
@@ -0,0 +1,158 @@
+
+/**
+ * Copyright (c) 2013-2018 OpenCFP
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ *
+ * @see https://github.com/opencfp/opencfp
+ */
+
+namespace OpenCFP\Http\Action\Security;
+
+use Cartalyst\Sentinel\Sentinel;
+use GuzzleHttp\Client;
+use OpenCFP\Domain\Services\AccountManagement;
+use OpenCFP\Infrastructure\Auth\SentinelUser;
+use OpenCFP\Infrastructure\Auth\UserNotFoundException;
+use Symfony\Component\HttpFoundation;
+use Symfony\Component\Routing;
+
+final class SsoRedirectAction
+{
+    /** @var AccountManagement */
+    private $accounts;
+
+    /** @var Sentinel */
+    private $sentinel;
+
+    /** @var Routing\Generator\UrlGeneratorInterface */
+    private $urlGenerator;
+
+    /** @var int */
+    private $clientId;
+
+    /** @var string */
+    private $clientSecret;
+
+    /** @var string */
+    private $redirectUri;
+
+    /** @var string */
+    private $resourceUri;
+
+    /** @var string */
+    private $tokenUrl;
+
+    /** @var Client */
+    private $httpClient;
+
+    /**
+     * SsoRedirectAction constructor.
+     *
+     * @param Sentinel                                $sentinel
+     * @param AccountManagement                       $accounts
+     * @param Routing\Generator\UrlGeneratorInterface $urlGenerator
+     * @param int                                     $clientId
+     * @param string                                  $clientSecret
+     * @param string                                  $redirectUri
+     * @param string                                  $resourceUri
+     * @param string                                  $tokenUrl
+     * @param Client                                  $httpClient
+     */
+    public function __construct(
+        Sentinel $sentinel,
+        AccountManagement $accounts,
+        Routing\Generator\UrlGeneratorInterface $urlGenerator,
+        int $clientId,
+        string $clientSecret,
+        string $redirectUri,
+        string $resourceUri,
+        string $tokenUrl,
+        Client $httpClient
+    ) {
+        $this->sentinel     = $sentinel;
+        $this->accounts     = $accounts;
+        $this->urlGenerator = $urlGenerator;
+        $this->clientId     = $clientId;
+        $this->clientSecret = $clientSecret;
+        $this->redirectUri  = $redirectUri;
+        $this->resourceUri  = $resourceUri;
+        $this->tokenUrl     = $tokenUrl;
+        $this->httpClient   = $httpClient;
+    }
+
+    /**
+     * @param HttpFoundation\Request $request
+     *
+     * @throws \OpenCFP\Infrastructure\Auth\UserExistsException
+     * @throws \Symfony\Component\Routing\Exception\RouteNotFoundException
+     * @throws \Symfony\Component\Routing\Exception\MissingMandatoryParametersException
+     * @throws \Symfony\Component\Routing\Exception\InvalidParameterException
+     * @throws \OpenCFP\Infrastructure\Auth\UserNotFoundException
+     * @throws \InvalidArgumentException
```

Comment:
Not sure if any of these exceptions are actually thrown (and not caught) anywhere in the scope of this method, can you check?